core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -467,7 +474,7 @@ private[spark] class TaskSchedulerImpl private[scheduler](
       taskState: TaskState,
       reason: TaskFailedReason): Unit = synchronized {
     taskSetManager.handleFailedTask(tid, taskState, reason)
-    if (!taskSetManager.isZombie && taskState != TaskState.KILLED) {
+    if (!taskSetManager.isZombie) {
```

Comment:
First, the assumption that reviveOffers is inexpensive is incorrect.
Secondly, task kill happens for a few reasons automatically - and can cause a bunch of tasks to be killed (for example, with speculative execution enabled, when one task finishes, the speculative task will be killed).

This can cause a request storm of revive offers to be launched.

Given this, the change is incorrect and should be reverted.